[Google Threat Intelligence] Add linux, malicious_network_infrastructure, malware, mobile and osx data streams

niraj-crest opened the pull request:
## Proposed commit message

This release includes five new data streams: `linux`, `malicious_network_infrastructure`, `malware`, `mobile`, and `osx`. Each data stream includes its own data collection logic, ingest pipeline, and associated dashboards and visualizations.

Fields are mapped to their corresponding ECS fields where applicable.

Test samples were derived from live logs and subsequently sanitized.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ti_google_threat_intelligence) directory.
- Run the following command to run tests. elastic-package test


<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

niraj-crest commented:
Hey @jamiehynds  & @andrewkroh 
Can anyone please add reviewers to this PR?

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

kcreddy commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24197)
* Commit: 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24197","number":24197,"commit":"83d29559126ce763fe36c18c7ed71df3db16f9dd"}],"number":24197}
buildkite-pr-comment-->

kcreddy reviewed (commented):
@niraj-crest, can you check if you can improve Sonar quality?
Looks like none of these pipelines are being covered (hence at 0% coverage):
<img width="1035" alt="Screenshot 2025-04-02 at 2 45 14 PM" src="https://github.com/user-attachments/assets/ed697a36-0119-4e41-ad84-2d4ea34866b8" />

niraj-crest commented:
> @niraj-crest, can you check if you can improve Sonar quality? Looks like none of these pipelines are being covered (hence at 0% coverage): <img alt="Screenshot 2025-04-02 at 2 45 14 PM" width="1035" src="https://private-user-images.githubusercontent.com/11301409/429422756-ed697a36-0119-4e41-ad84-2d4ea34866b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM2NzEwMjUsIm5iZiI6MTc0MzY3MDcyNSwicGF0aCI6Ii8xMTMwMTQwOS80Mjk0MjI3NTYtZWQ2OTdhMzYtMDExOS00ZTQxLWFkODQtMmQ0ZWEzNDg2NmI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDAzVDA4NTg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OTlhOWVkYjZiZThkZDgxOWQ3MGY0YTkxZDI0ODJiZmRiMzE3MTExMDY0ZGVkMzViMmVjYzkxMWYxNjE1NTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.a-gxX4cTT64aZV6Df5Y61sCUemkXqFNJlgwgmiVVPJI">

@kcreddy Have removed the pipelines and transforms as they are covered in this [PR](https://github.com/elastic/integrations/pull/13189).

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24343)
* Commit: d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8

### History
* :green_heart: [Build #24197](https://buildkite.com/elastic/integrations/builds/24197) succeeded 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24343","number":24343,"commit":"d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24197","number":24197,"commit":"83d29559126ce763fe36c18c7ed71df3db16f9dd"}],"number":24343}
buildkite-pr-comment-->

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+      }.format_query()
+    ).with({
+      "Header": {
+        "x-apikey": [string(state.access_token)]
```
```suggestion
        "x-apikey": [state.access_token]
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+        "x-apikey": [string(state.access_token)]
+      }
+    }).do_request().as(resp, resp.StatusCode == 200 ?
+      bytes(resp.Body).decode_json().as(body, {
```
```suggestion
      resp.Body.decode_json().as(body, {
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+config_version: 2
+interval: {{interval}}
+{{#if enable_request_tracer}}
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
+resource.tracer.maxbackups: 5
+{{/if}}
```
```suggestion
resource.tracer:
  enabled: {{enable_request_tracer}}
  filename: "../../logs/cel/http-request-trace-*.ndjson"
  maxbackups: 5
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    request(
+      "GET",
+      state.url.trim_right("/") + "/api/v3/threat_lists/linux/"+state.start_time+"?"+{
+        ?"query": has(state.query) && state.query != null ? optional.of([state.query]) : optional.none(),
```
When will `state.query` be present but `null`? I can't see that being done anywhere.
```suggestion
        ?"query": has(state.query) ? optional.of([state.query]) : optional.none(),
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    }).do_request().as(resp, resp.StatusCode == 200 ?
+      bytes(resp.Body).decode_json().as(body, {
+        "events": (
+            has(body.iocs) && size(body.iocs) > 0
+          ?
```
```suggestion
          has(body.iocs) && size(body.iocs) > 0 ?
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,97 @@
+    request(
+      "GET",
+      state.url.trim_right("/") + "/api/v3/threat_lists/malicious-network-infrastructure/"+state.start_time+"?"+{
+        ?"query": has(state.query) && state.query != null ? optional.of([state.query]) : optional.none(),
```
```suggestion
        ?"query": has(state.query) ? optional.of([state.query]) : optional.none(),
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,97 @@
+      }.format_query()
+    ).with({
+      "Header": {
+        "x-apikey": [string(state.access_token)]
```
```suggestion
        "x-apikey": [state.access_token]
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,97 @@
+    }).do_request().as(resp, resp.StatusCode == 200 ?
+      bytes(resp.Body).decode_json().as(body, {
+        "events": (
+            has(body.iocs) && size(body.iocs) > 0
+          ?
```
```suggestion
          has(body.iocs) && size(body.iocs) > 0 ?
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,97 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": "GET:"+(
```
```suggestion
            "message": "GET "+state.url.trim_right("/")+"/api/v3/threat_lists/malicious-network-infrastructure/:"+(
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,97 @@
+config_version: 2
+interval: {{interval}}
+{{#if enable_request_tracer}}
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
+resource.tracer.maxbackups: 5
+{{/if}}
```
```suggestion
resource.tracer:
  enabled: {{enable_request_tracer}}
  filename: "../../logs/cel/http-request-trace-*.ndjson"
  maxbackups: 5
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malware/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+config_version: 2
+interval: {{interval}}
+{{#if enable_request_tracer}}
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
+resource.tracer.maxbackups: 5
+{{/if}}
```
```suggestion
resource.tracer:
  enabled: {{enable_request_tracer}}
  filename: "../../logs/cel/http-request-trace-*.ndjson"
  maxbackups: 5
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malware/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    request(
+      "GET",
+      state.url.trim_right("/") + "/api/v3/threat_lists/malware/"+state.start_time+"?"+{
+        ?"query": has(state.query) && state.query != null ? optional.of([state.query]) : optional.none(),
```
```suggestion
        ?"query": has(state.query) ? optional.of([state.query]) : optional.none(),
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malware/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+      }.format_query()
+    ).with({
+      "Header": {
+        "x-apikey": [string(state.access_token)]
```
```suggestion
        "x-apikey": [state.access_token]
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malware/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+            "message": resp.StatusCode == 403 ?
+              "Permission denied. Malware event collection is restricted to Enterprise or Enterprise Plus subscriptions. Contact your administrator or upgrade your subscription to enable this feature."
+            :
+              "GET:"+(
```
```suggestion
              "GET "+state.url.trim_right("/")+"/api/v3/threat_lists/malware/:"+(
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/malware/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": resp.StatusCode == 403 ?
```
I don't think this is a good idea.

efd6 commented on packages/ti_google_threat_intelligence/data_stream/mobile/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    request(
+      "GET",
+      state.url.trim_right("/") + "/api/v3/threat_lists/mobile/"+state.start_time+"?"+{
+        ?"query": has(state.query) && state.query != null ? optional.of([state.query]) : optional.none(),
```
```suggestion
        ?"query": has(state.query) ? optional.of([state.query]) : optional.none(),
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/mobile/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    }).do_request().as(resp, resp.StatusCode == 200 ?
+      bytes(resp.Body).decode_json().as(body, {
+        "events": (
+            has(body.iocs) && size(body.iocs) > 0
+          ?
```
```suggestion
          has(body.iocs) && size(body.iocs) > 0 ?
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/mobile/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": resp.StatusCode == 403 ?
```
I don't think this is a good idea.

efd6 commented on packages/ti_google_threat_intelligence/data_stream/mobile/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+            "message": resp.StatusCode == 403 ?
+              "Permission denied. Mobile event collection is restricted to Enterprise Plus subscriptions. Contact your administrator or upgrade your subscription to enable this feature."
+            :
+              "GET:"+(
```
```suggestion
              "GET "+state.url.trim_right("/")+"/api/v3/threat_lists/mobile/:"+(
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/mobile/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+config_version: 2
+interval: {{interval}}
+{{#if enable_request_tracer}}
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
+resource.tracer.maxbackups: 5
+{{/if}}
```
```suggestion
resource.tracer:
  enabled: {{enable_request_tracer}}
  filename: "../../logs/cel/http-request-trace-*.ndjson"
  maxbackups: 5
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+config_version: 2
+interval: {{interval}}
+{{#if enable_request_tracer}}
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
+resource.tracer.maxbackups: 5
+{{/if}}
```
```suggestion
resource.tracer:
  enabled: {{enable_request_tracer}}
  filename: "../../logs/cel/http-request-trace-*.ndjson"
  maxbackups: 5
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    request(
+      "GET",
+      state.url.trim_right("/") + "/api/v3/threat_lists/osx/"+state.start_time+"?"+{
+        ?"query": has(state.query) && state.query != null ? optional.of([state.query]) : optional.none(),
```
```suggestion
        ?"query": has(state.query) ? optional.of([state.query]) : optional.none(),
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+      }.format_query()
+    ).with({
+      "Header": {
+        "x-apikey": [string(state.access_token)]
```
```suggestion
        "x-apikey": [state.access_token]
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+    }).do_request().as(resp, resp.StatusCode == 200 ?
+      bytes(resp.Body).decode_json().as(body, {
+        "events": (
+            has(body.iocs) && size(body.iocs) > 0
+          ?
```
```suggestion
          has(body.iocs) && size(body.iocs) > 0 ?
```

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": resp.StatusCode == 403 ?
```
I don't think this is a good idea.

efd6 commented on packages/ti_google_threat_intelligence/data_stream/osx/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+            "message": resp.StatusCode == 403 ?
+              "Permission denied. OS X event collection is restricted to Enterprise Plus subscriptions. Contact your administrator or upgrade your subscription to enable this feature."
+            :
+              "GET:"+(
```
```suggestion
              "GET "+state.url.trim_right("/")+"/api/v3/threat_lists/osx/:"+(
```

efd6 reviewed (commented):
Reviewed CEL code only at this stage.

kcreddy commented:
@niraj-crest, please add all of @efd6 review suggestions to https://github.com/elastic/integrations/pull/13189 as well. Thanks!

niraj-crest commented:
> @niraj-crest, please add all of @efd6 review suggestions to #13189 as well. Thanks!

Sure, currently working on it.

niraj-crest commented:
Waiting for reply on this [comment](https://github.com/elastic/integrations/pull/13189#discussion_r2041353605). Once we have a finalized approach, I will push all the changes accordingly.

niraj-crest commented:
@kcreddy I have addressed review comments, Please run tests.

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": resp.StatusCode == 403 ?
```
I'm still not sure that this is the right place for this. But leaving for @kcreddy to decide. If it stays, please add spaces after each of the colons in the text for 403.

kcreddy replied:
I would second that. Please add the [generic error part ](https://github.com/elastic/integrations/blob/main/packages/proofpoint_itm/data_stream/report/agent/stream/cel.yml.hbs#L51-L61)and if it has the body, it would be added into `error.message`. 

If you would like to provide more details such as this and possible reasons for 403, you can add it to troubleshooting section inside README.

kcreddy commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/26103)
* Commit: cd98086cb9de7d2d04d6e87a238464b6a1ad8778

### History
* :green_heart: [Build #24343](https://buildkite.com/elastic/integrations/builds/24343) succeeded d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8
* :green_heart: [Build #24197](https://buildkite.com/elastic/integrations/builds/24197) succeeded 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26103","number":26103,"commit":"cd98086cb9de7d2d04d6e87a238464b6a1ad8778"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24343","number":24343,"commit":"d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24197","number":24197,"commit":"83d29559126ce763fe36c18c7ed71df3db16f9dd"}],"number":26103}
buildkite-pr-comment-->

efd6 commented on packages/ti_google_threat_intelligence/data_stream/linux/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,93 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": "GET " + state.url.trim_right("/") + "/api/v3/threat_lists/linux/" + state.start_time + ":" + (
```
We don't provide this level of detail for any other integration. I don't think that we should start doing it here without discussion.

efd6 commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/26154)
* Commit: 86f99e342b6af4efa8c88857b6888a06500b25ae

### History
* :green_heart: [Build #26103](https://buildkite.com/elastic/integrations/builds/26103) succeeded cd98086cb9de7d2d04d6e87a238464b6a1ad8778
* :green_heart: [Build #24343](https://buildkite.com/elastic/integrations/builds/24343) succeeded d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8
* :green_heart: [Build #24197](https://buildkite.com/elastic/integrations/builds/24197) succeeded 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26154","number":26154,"commit":"86f99e342b6af4efa8c88857b6888a06500b25ae"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26103","number":26103,"commit":"cd98086cb9de7d2d04d6e87a238464b6a1ad8778"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24343","number":24343,"commit":"d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24197","number":24197,"commit":"83d29559126ce763fe36c18c7ed71df3db16f9dd"}],"number":26154}
buildkite-pr-comment-->

kcreddy reviewed (commented):
Please apply latest review comments from https://github.com/elastic/integrations/pull/13189#pullrequestreview-2909576304

jamiehynds commented on packages/ti_google_threat_intelligence/manifest.yml:
```diff
@@ -0,0 +1,85 @@
+              #    -----END CERTIFICATE-----
+owner:
+  github: elastic/security-service-integrations
+  type: community
```
Community is generally for integrations built by users which doesn't fit this case. "Partner" is probably the best fit in this case, as a technology partner (Google) has funded and built the integration.

niraj-crest commented:
Hello @kcreddy, @efd6,
We have Applied all the Comments of this PR https://github.com/elastic/integrations/pull/13189.

Can you please start Reviewing it.

kcreddy commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/27697)
* Commit: 86930fb9af5a9b3d78b416f481ad55a7534b4dc3

### History
* :green_heart: [Build #26520](https://buildkite.com/elastic/integrations/builds/26520) succeeded 5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe
* :green_heart: [Build #26154](https://buildkite.com/elastic/integrations/builds/26154) succeeded 86f99e342b6af4efa8c88857b6888a06500b25ae
* :green_heart: [Build #26103](https://buildkite.com/elastic/integrations/builds/26103) succeeded cd98086cb9de7d2d04d6e87a238464b6a1ad8778
* :green_heart: [Build #24343](https://buildkite.com/elastic/integrations/builds/24343) succeeded d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8
* :green_heart: [Build #24197](https://buildkite.com/elastic/integrations/builds/24197) succeeded 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/27697","number":27697,"commit":"86930fb9af5a9b3d78b416f481ad55a7534b4dc3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26520","number":26520,"commit":"5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26154","number":26154,"commit":"86f99e342b6af4efa8c88857b6888a06500b25ae"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26103","number":26103,"commit":"cd98086cb9de7d2d04d6e87a238464b6a1ad8778"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24343","number":24343,"commit":"d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24197","number":24197,"commit":"83d29559126ce763fe36c18c7ed71df3db16f9dd"}],"number":27697}
buildkite-pr-comment-->

kcreddy commented on packages/ti_google_threat_intelligence/_dev/build/docs/README.md:
```diff
@@ -14,6 +14,11 @@ The Threat List API provides the following types of threat feeds:
 - **First Stage Delivery Vectors**
 - **Infostealers**
 - **Internet of Things (IoT)**
+- **Linux**
+- **Malicious Network Infrastructure**
+- **Malware**
+- **Mobile**
+- **OS X**
```
Suggest you add a one-liner statement or add a vendor documentation link to each of these feeds if exists.

niraj-crest replied:
Thanks for the suggestion! However, there is currently no separate vendor documentation available for each individual threat feed.
That said, the [Get Hourly Threat List](https://gtidocs.virustotal.com/reference/get-hourly-threat-list) endpoint provides a table with detailed information covering all of these feeds, and we've already included that reference.

kcreddy commented on packages/ti_google_threat_intelligence/data_stream/malicious_network_infrastructure/sample_event.json:
```diff
@@ -0,0 +1,117 @@
+    ],
+    "threat": {
+        "feed": {
+            "name": "Malicious Network Infra"
```
```suggestion
            "name": "Malicious Network Infrastructure"
```
to be consistent

kcreddy commented on packages/ti_google_threat_intelligence/data_stream/malware/sample_event.json:
```diff
@@ -0,0 +1,117 @@
+    ],
+    "threat": {
+        "feed": {
+            "name": "Malware"
```
You could also add [`threat.feed.dashboard_id`](https://www.elastic.co/docs/reference/ecs/ecs-threat#field-threat-feed-dashboard-id)

kcreddy commented on packages/ti_google_threat_intelligence/elasticsearch/transform/domain_ioc/fields/linux_fields.yml:
For all the transforms' `ecs.yml`, can you also add following change: https://github.com/elastic/integrations/pull/14290 ?. Looks like we missed this field and it dynamically gets mapped to different type.

niraj-crest commented on packages/ti_google_threat_intelligence/data_stream/linux/_dev/test/pipeline/test-linux.log-expected.json:
```diff
@@ -0,0 +1,371 @@
+            ],
+            "threat": {
+                "feed": {
+                    "name": "Linux"
```
Sure,
![image](https://github.com/user-attachments/assets/ffaa84dc-f970-40e7-9490-78fa701b4bb5)

Would you Suggest us to update the `threat.feed.name` for all the data streams.

kcreddy replied:
> Would you Suggest us to update the threat.feed.name for all the data streams.

@niraj-crest , thanks for the screenshot. Yes, that would be nice.

kcreddy reviewed (commented):
LGTM after https://github.com/elastic/integrations/pull/13190#discussion_r2184586837 and https://github.com/elastic/integrations/pull/13190#discussion_r2184584698

kcreddy commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/28071)
* Commit: d5160c498fce16f0cd0d5bd63b7be342dccad699

### History
* :green_heart: [Build #27697](https://buildkite.com/elastic/integrations/builds/27697) succeeded 86930fb9af5a9b3d78b416f481ad55a7534b4dc3
* :green_heart: [Build #26520](https://buildkite.com/elastic/integrations/builds/26520) succeeded 5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe
* :green_heart: [Build #26154](https://buildkite.com/elastic/integrations/builds/26154) succeeded 86f99e342b6af4efa8c88857b6888a06500b25ae
* :green_heart: [Build #26103](https://buildkite.com/elastic/integrations/builds/26103) succeeded cd98086cb9de7d2d04d6e87a238464b6a1ad8778
* :green_heart: [Build #24343](https://buildkite.com/elastic/integrations/builds/24343) succeeded d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8
* :green_heart: [Build #24197](https://buildkite.com/elastic/integrations/builds/24197) succeeded 83d29559126ce763fe36c18c7ed71df3db16f9dd

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/28071","number":28071,"commit":"d5160c498fce16f0cd0d5bd63b7be342dccad699"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/27697","number":27697,"commit":"86930fb9af5a9b3d78b416f481ad55a7534b4dc3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26520","number":26520,"commit":"5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26154","number":26154,"commit":"86f99e342b6af4efa8c88857b6888a06500b25ae"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26103","number":26103,"commit":"cd98086cb9de7d2d04d6e87a238464b6a1ad8778"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24343","number":24343,"commit":"d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8"}],"number":28071}
buildkite-pr-comment-->

kcreddy commented:
/test

kcreddy reviewed (approved):
LGTM, but please wait for @efd6 approval before merging.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/28186)
* Commit: d82006eec81e075e743e452fe328c9c5bd3261db

### History
* :green_heart: [Build #28071](https://buildkite.com/elastic/integrations/builds/28071) succeeded d5160c498fce16f0cd0d5bd63b7be342dccad699
* :green_heart: [Build #27697](https://buildkite.com/elastic/integrations/builds/27697) succeeded 86930fb9af5a9b3d78b416f481ad55a7534b4dc3
* :green_heart: [Build #26520](https://buildkite.com/elastic/integrations/builds/26520) succeeded 5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe
* :green_heart: [Build #26154](https://buildkite.com/elastic/integrations/builds/26154) succeeded 86f99e342b6af4efa8c88857b6888a06500b25ae
* :green_heart: [Build #26103](https://buildkite.com/elastic/integrations/builds/26103) succeeded cd98086cb9de7d2d04d6e87a238464b6a1ad8778
* :green_heart: [Build #24343](https://buildkite.com/elastic/integrations/builds/24343) succeeded d8dd11d5bb2155f92943dbafd64cfe6036bfe0a8

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/28186","number":28186,"commit":"d82006eec81e075e743e452fe328c9c5bd3261db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/28071","number":28071,"commit":"d5160c498fce16f0cd0d5bd63b7be342dccad699"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/27697","number":27697,"commit":"86930fb9af5a9b3d78b416f481ad55a7534b4dc3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26520","number":26520,"commit":"5ebdb1e93ad2b75a04c035e2e51a30a6fdae2cfe"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26154","number":26154,"commit":"86f99e342b6af4efa8c88857b6888a06500b25ae"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/26103","number":26103,"commit":"cd98086cb9de7d2d04d6e87a238464b6a1ad8778"}],"number":28186}
buildkite-pr-comment-->

niraj-crest commented:
Hello @efd6,
We’ve addressed all the comments, could you please take look at the PR?
If everything looks good, we’d appreciate if we could proceed with merging.

Thank you!

efd6 commented on packages/ti_google_threat_intelligence/data_stream/cryptominer/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -64,6 +64,10 @@ processors:
+  - set:
+      field: threat.feed.dashboard_id
+      tag: set_threat_feed_dashboard_id
+      value: ['ti_google_threat_intelligence-0b0fb6b4-d250-4e31-a56a-bb872e4c7c4a', 'ti_google_threat_intelligence-9e8de699-a623-4a1b-9f63-7d641116f531', 'ti_google_threat_intelligence-95187e5c-b4a2-45ad-b6a4-d6ce68e1f43e']
```
Does this not work?
```
  - set:
      field: threat.feed.dashboard_id
      tag: set_threat_feed_dashboard_id
      value:
        - ti_google_threat_intelligence-0b0fb6b4-d250-4e31-a56a-bb872e4c7c4a
        - ti_google_threat_intelligence-9e8de699-a623-4a1b-9f63-7d641116f531
        - ti_google_threat_intelligence-95187e5c-b4a2-45ad-b6a4-d6ce68e1f43e
```

efd6 reviewed (commented):
Nit in https://github.com/elastic/integrations/pull/13190#discussion_r2194143774

efd6 commented:
@niraj-crest Please use the PR template that exists in the repo so that we have a proposed commit message. I'll put one together for now.
